monitoring/uss_qualifier/suites/faa/uft/design/README.md

```diff
@@ -0,0 +1,79 @@
+
+The main idea behind the tests is that mock_uss will consume and validate all the requests and responses from USS-under-test.
+
+Note - As different USSes have different implementations, it could happen that you might need to create a subscription in the area of the flights.
+
+## Steps to run the test
+
+1. Set your UssQualifier Interface implementation url in the [configuration file ](../../../../configurations/dev/faa/uft/local_message_signing.yaml )to run the UFT message signing tests.
+The property to set is `resources.resource_declarations.flight_planners.specification.flight_planners.participant_id`
+2. Run DSS and dummy-oauth using the script [run_locally.sh](../../../../../../build/dev/run_locally.sh)
+    ```bash
+    ./run_locally.sh
+    ```
+3. Run mock_uss using the script [run_locally_scdsc.sh](../../../../../mock_uss/run_locally_scdsc.sh)
+    ```bash
+   ./run_locally_scdsc.sh
+    ```
+   Note - This script has an environment variable MESSAGE_SIGNING. It is set to true to turn message signing validation on in mock uss.
+4. Run well-known server with public key for mock_uss. [run_well_known.sh]
+    ```bash
+   ./run_well_known.sh
+   ```
+5. Prepare your USS to run with
+   1. dummy-oauth for getting tokens. Depending on your setup - localhost(http://localhost:8085/token) or dockerized (http://host.docker.internal:8085/token). The server has a GET and POST endpoint for getting tokens.
+   2. DSS - Depending on your setup - localhost(http://localhost:8082/dss) or dockerized (http://host.docker.internal:8082/dss)
+6. Run Uss Qualifier Interface for your USS.
+7. Run USS Qualifier tests using script [run_locally.sh](../../../../../uss_qualifier/run_locally.sh) with config
+    ```bash
+   ./run_locally.sh configurations.dev.faa.uft.local_message_signing
+   ```
+
+## Results
+
+The USS Qualifier tests run produces two reports.
+1. SCD tests report is generated under [uss_qualifier](../../../../../uss_qualifier).
+The report file name is - report.json
+2. Message signing test report is generated under [mock_uss/report]
+The report file name is - report_mock_uss_scdsc_messagesigning.json.
+
+There will be 3 runs of tests, with the above two reports for each run.
+
+### Positive tests -
+A valid set of private/public keys will be provided under [keys folder]
+A USS should pass all the uss_qualifier tests in this suite.
+As well as the message_signing report should have no issues reported. A USS will provide both of these reports.
+The message_signing report includes interactions and issues between the mock_uss and the USS-under-test.
+No issues indicate the USS-under-test message-signed all its requests and responses.
+
+For analysis of the message signing reports, please refer to [Analyzing the message signing report in README.md]
+
+### Negative tests -
+Replace the private/public keys with invalid key pair, the message signing by mock_uss will be invalid.
```

Comment:
This seems like a lot of manual work for an "automated" test; why wouldn't we expose a control endpoint for mock_uss that allows uss_qualifier to specify a keypair to use and then have 3 different test scenarios for Positive, Negative, and No message signing?  The first MVP could of course provide support for only one of these scenarios without that control endpoint, and then the test could be built out to support that control endpoint as a second phase.

Reply:
I agree with that. Currently, to deliver for the scheduled events for UFT, we were able to make the flow for positive tests only. We need to put in more time and effort to get the negative tests in. However, till we are not ready with those tests, we thought that for the negative tests, we can ask the USSes to run it again with wrong key pairs. But, you are right, if time was not a constraint, we would not ask for these manual steps.